File: tmdb-core/src/main/java/dev/reuss/tmdb/core/exception/TmdbRateLimitException.java
package dev.reuss.tmdb.core.exception;

/**
 * Exception thrown when TMDB rejects a request because the rate limit was exceeded.
 */
public final class TmdbRateLimitException extends TmdbApiException {

    /**
     * Creates a rate-limit API exception.
     *
     * @param message error message
     * @param httpStatus HTTP status code
     * @param tmdbStatusCode TMDB status code, or {@code 0} if unavailable
     * @param responseBody raw response body
     */
    public TmdbRateLimitException(String message, int httpStatus, int tmdbStatusCode, String responseBody) {
        super(message, httpStatus, tmdbStatusCode, responseBody);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/core/exception/TmdbRateLimitException.kt
package dev.reuss.tmdb.core.exception

/**
 * Exception thrown when TMDB rejects a request because the rate limit was exceeded.
 *
 * @property message error message
 * @property httpStatus HTTP status code
 * @property tmdbStatusCode TMDB status code, or `0` if unavailable
 * @property responseBody raw response body
 */
class TmdbRateLimitException(
    message: String,
    httpStatus: Int,
    tmdbStatusCode: Int,
    responseBody: String
) : TmdbApiException(
    message,
    httpStatus,
    tmdbStatusCode,
    responseBody
)